dvc/repo/experiments/executor/base.py

```diff
@@ -623,6 +627,18 @@ def _repro_dvc(  # noqa: C901
             if cls.QUIET:
                 dvc.scm_context.quiet = cls.QUIET
             old_cwd = os.getcwd()
+
+            if copy_paths:
+                for path in copy_paths:
+                    if os.path.isfile(path):
+                        shutil.copy(
+                            os.path.realpath(path), os.path.join(dvc.root_dir, path)
```

Comment:
Since defining `copy-paths` expected format is in our hands, I though that making it relative to the `dvc.root_dir` made sense for simplicity

Reply:
Not sure I follow. When will it matter?

Also, should we be using [`follow_symlinks=False`](https://docs.python.org/3/library/shutil.html#shutil.copy) (see https://github.com/iterative/cse/issues/99)?

Reply:
> Not sure I follow. When will it matter?

It would matter depending on how the paths are defined/used inside the `dvc.yaml` vs how they are provided as an argument to `copy-paths`. 

> Also, should we be using [follow_symlinks=False](https://docs.python.org/3/library/shutil.html#shutil.copy) (see https://github.com/iterative/cse/issues/99)?

[`os.path.realpath`](https://docs.python.org/3/library/os.path.html#os.path.realpath) is used so I think it should not be needed (https://github.com/iterative/cse/issues/99 works)

Reply:
> [`os.path.realpath`](https://docs.python.org/3/library/os.path.html#os.path.realpath) is used so I think it should not be needed ([iterative/cse#99](https://github.com/iterative/cse/issues/99) works)

Does it create an actual copy? I think the desired behavior would be to create another symlink to the same source.

Reply:
We can't use `follow_symlinks=False`. If the original link uses a relative path, the newly copied symlink will point to the wrong location

(`shutil.copy(..., follow_symlinks=False)` creates a new link using the same relative path, not a new link pointing to the resolved absolute path of the original target)

Reply:
I don't think we should be using links here anyways. If the file ends up being modified by the experiment pipeline, the tempdir executor should be modifying a completely independent copy of that file. (We should not be writing to the original file via a symlink)

Reply:
> I don't think we should be using links here anyways. If the file ends up being modified by the experiment pipeline, the tempdir executor should be modifying a completely independent copy of that file. (We should not be writing to the original file via a symlink)

Unless I am missing something, this is already the behavior present in the P.R:

- `os.path.realpath` resolves all symlinks and give the "true" path.
- `shutil.copy` / `shutil.copytree` make actual copies into the tempdir.

Reply:
Yes, the PR should be fine as-is, I was just commenting in response to @dberenbaum's question

Reply:
Okay, thanks for the explanation. It doesn't fully resolve https://github.com/iterative/cse/issues/99 in that case since they want to create links to their large data dependency, but it's not a blocker.